src/MiniCover/Reports/HtmlReport.cs

```diff
@@ -55,32 +55,30 @@ protected override void WriteDetailedReport(InstrumentationResult result, IDicti
                     htmlWriter.WriteLine("<html>");
                     htmlWriter.WriteLine("<body style=\"font-family: sans-serif;\">");
 
-                    var instrumentedLineNumbers = kvFile.Value.Instructions
-                        .SelectMany(i => Enumerable.Range(i.StartLine, i.EndLine - i.StartLine + 1))
-                        .Distinct()
-                        .ToArray();
-
-                    var hitInstructions = kvFile.Value.Instructions.Where(h => hits.Contains(h.Id)).ToArray();
-                    var coveredLineNumbers = hitInstructions
-                        .SelectMany(i => Enumerable.Range(i.StartLine, i.EndLine - i.StartLine + 1))
-                        .Distinct()
-                        .ToArray();
+                    var uncoveredLineNumbers = new HashSet<int>();
+                    foreach(var i in kvFile.Value.Instructions)
```

Comment:
Can you please merge both foreachs and add curly braces?
```
foreach(var i in kvFile.Value.Instructions) {
  if (hits.Contains(i.Id)) { 
   ...
  } else {
   ...
  }
}
```

Reply:
You are right, this will be more understandable. And a tiny bit faster :zap: